src/Microsoft.Health.Fhir.Shared.Core/Features/Search/SearchOptionsFactory.cs

```diff
@@ -90,6 +90,13 @@ public SearchOptions Create(string compartmentType, string compartmentId, string
                     // Query parameter with empty value is not supported.
                     unsupportedSearchParameters.Add(query);
                 }
+                else if (string.Compare(query.Item1, KnownQueryParameterNames.Total, StringComparison.OrdinalIgnoreCase) == 0)
+                {
+                    if (Enum.TryParse<TotalType>(query.Item2, true, out var totalType))
```

Comment:
If it fails, should it throw BadRequest? Also, if estimate is not supported, should we throw error indicating it's not supported?

Reply:
Good question. Since it wasn't specified in the spec, I followed the implementation of `_pretty`  (which falls back to the default setting of `false` when an invalid/incorrect value is passed).

Looking at the implementations of `_summary` and `_format` now, it seems like your suggestion makes more sense. Do you think the lack of error messages for invalid `_pretty` values should be logged as a bug?

Reply:
Logged a [bug](https://microsofthealth.visualstudio.com/Health/_workitems/edit/70807) for the _pretty parameter and added invalid input handling for the _total parameter in 9830b25.